This patch adds the necessary changes to create and manage tags shareable on
all screens when using the awesome window manager.

The library and all relevant documentation can be found at
https://github.com/Drauthius/awesome-sharedtags

--- orig/rc.lua	2016-01-01 13:45:26.000000000 +0100
+++ sharedtags/rc.lua	2016-01-05 20:09:21.428836239 +0100
@@ -36,6 +36,11 @@
 end
 -- }}}
 
+-- {{{ Local extensions
+local sharedtags = require("sharedtags")
+-- }}}
+
+
 -- {{{ Variable definitions
 -- Themes define colours, icons, font and wallpapers.
 beautiful.init("/usr/share/awesome/themes/default/theme.lua")
@@ -79,12 +84,16 @@
 -- }}}
 
 -- {{{ Tags
--- Define a tag table which hold all screen tags.
-tags = {}
-for s = 1, screen.count() do
-    -- Each screen has its own tag table.
-    tags[s] = awful.tag({ 1, 2, 3, 4, 5, 6, 7, 8, 9 }, s, layouts[1])
-end
+-- Define a tag table shared among all screens.
+local tags = sharedtags({
+    { name = "main", layout = layouts[2] },
+    { name = "www", layout = layouts[10] },
+    { name = "chat", screen = 2, layout = layouts[1] },
+    { name = "game", layout = layouts[1] },
+    { layout = layouts[2] },
+    { layout = layouts[2] },
+    { screen = 2, layout = layouts[2] }
+})
 -- }}}
 
 -- {{{ Menu
@@ -302,26 +311,24 @@
         -- View tag only.
         awful.key({ modkey }, "#" .. i + 9,
                   function ()
-                        local screen = mouse.screen
-                        local tag = awful.tag.gettags(screen)[i]
+                        local tag = tags[i]
                         if tag then
-                           awful.tag.viewonly(tag)
+                           sharedtags.viewonly(tag)
                         end
                   end),
         -- Toggle tag.
         awful.key({ modkey, "Control" }, "#" .. i + 9,
                   function ()
-                      local screen = mouse.screen
-                      local tag = awful.tag.gettags(screen)[i]
+                      local tag = tags[i]
                       if tag then
-                         awful.tag.viewtoggle(tag)
+                         sharedtags.viewtoggle(tag)
                       end
                   end),
         -- Move client to tag.
         awful.key({ modkey, "Shift" }, "#" .. i + 9,
                   function ()
                       if client.focus then
-                          local tag = awful.tag.gettags(client.focus.screen)[i]
+                          local tag = tags[i]
                           if tag then
                               awful.client.movetotag(tag)
                           end
@@ -331,7 +338,7 @@
         awful.key({ modkey, "Control", "Shift" }, "#" .. i + 9,
                   function ()
                       if client.focus then
-                          local tag = awful.tag.gettags(client.focus.screen)[i]
+                          local tag = tags[i]
                           if tag then
                               awful.client.toggletag(tag)
                           end
@@ -365,9 +372,9 @@
       properties = { floating = true } },
     { rule = { class = "gimp" },
       properties = { floating = true } },
-    -- Set Firefox to always map on tags number 2 of screen 1.
+    -- Set Firefox to always map on tag number 2.
     -- { rule = { class = "Firefox" },
-    --   properties = { tag = tags[1][2] } },
+    --   properties = { tag = tags[2] } }, -- tag = tags["www"] works as well
 }
 -- }}}
 
